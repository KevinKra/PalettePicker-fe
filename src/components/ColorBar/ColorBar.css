.color-block {
  display: flex;
  flex-direction: column;
  flex: 1;
  height: 100%;
  /* border: 1px solid red; */
}

.color-block:hover {
  cursor: pointer;
}

.lock-button {
  border: 1px solid transparent;
  transition: 0.15s all ease-in;
  opacity: 0;
  margin: 0;
  padding: 0;
  height: 0%;
}

.lock-button:focus {
  outline: none;
}

i {
  color: rgb(56, 56, 56);
}

.color-block:hover > button {
  cursor: pointer;
  opacity: 1;
  height: 15%;
}
.color-bar {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  transition: 0.07s border ease-in-out;
}

.color-block:hover > .color-bar {
  border-top: none;
  border-bottom: 5px solid black;
  /* box-shadow: -1px -6px 7px 2px rgba(0, 0, 0, 0.74); */
}

.color-bar > p {
  text-align: center;
}

.color-bar-text-vertical {
  position: absolute;
  transform: rotate(-90deg) translateX(2vw);
  width: 100%;
}

@keyframes move {
  0% {
    transform: translateY(-150%);
  }
  100% {
    transform: translateY(0%);
  }
}
