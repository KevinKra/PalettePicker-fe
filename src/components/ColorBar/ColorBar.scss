@import "../../_styles/main.scss";

.color-block {
	display: flex;
	flex-direction: column;
	flex: 1;
	height: 100.5%;
}

.color-block:hover {
	cursor: pointer;
}

.lock-button {
	border: 1px solid transparent;
	transition: 0.15s all ease-in;
	visibility: hidden;
	opacity: 0;
	margin: 0;
	padding: 0;
	height: 0%;
	background-color: rgb(40, 48, 56);
}

.lock-button:focus {
	outline: none;
	/* background-color: red; */
}

i {
	color: white;
}

.color-block:hover > button {
	visibility: visible;
	cursor: pointer;
	opacity: 1;
	height: 15%;
}
.color-bar {
	position: relative;
	display: flex;
	justify-content: center;
	align-items: center;
	height: 100%;
	transition: 0.07s border ease-in-out;
}

.color-block:hover > .color-bar {
	border-top: none;
	border-bottom: 5px solid black;
	/* box-shadow: -1px -6px 7px 2px rgba(0, 0, 0, 0.74); */
}

.color-bar > p {
	text-align: center;
}

.color-bar-text-vertical {
	position: absolute;
	transform: rotate(-90deg) translateX(2vw);
	width: 100%;
}

@keyframes move {
	0% {
		transform: translateY(-150%);
	}
	100% {
		transform: translateY(0%);
	}
}
