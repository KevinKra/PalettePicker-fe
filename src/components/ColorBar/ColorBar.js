import React, { Component } from "react";
import "./ColorBar.css";

export default class ColorBar extends Component {
  state = {
    locked: false
  };

  lockColor = () => {
    this.props.lockColor(this.props.color);
    this.setState({ locked: true });
  };

  unlockColor = () => {
    this.props.unlockColor(this.props.color);
    this.setState({ locked: false });
  };

  render() {
    const colorBarStyle = {
      backgroundColor: `${this.props.color}`
    };
    const defaultStyle = {
      transform: "translateY(-150%)",
      animation: `move 0.7s ease-in ${this.props.number / 8}s forwards`
    };
    const lockedBarStyle = {
      backgroundColor: `${this.props.color}`,
      borderBottom: "5px solid black"
    };
    const lockedStyle = {
      visibility: "visible",
      height: "15%",
      opacity: 1
    };
    return (
      <div className="color-block" style={defaultStyle}>
        <div
          className={`color-bar color-${this.props.number + 1}`}
          style={this.state.locked ? lockedBarStyle : colorBarStyle}
        >
          <p className={this.props.vRotate && "color-bar-text-vertical"}>
            {this.props.color.toUpperCase()}
          </p>
        </div>
        <button
          className="lock-button"
          onClick={this.state.locked ? this.unlockColor : this.lockColor}
          style={this.state.locked ? lockedStyle : null}
        >
          {this.state.locked ? (
            <i className="fas fa-2x fa-lock" />
          ) : (
            <i className="fas fa-2x fa-unlock-alt" />
          )}
        </button>
      </div>
    );
  }
}
