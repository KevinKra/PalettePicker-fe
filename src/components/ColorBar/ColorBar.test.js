import React from "react";
import ColorBar from "./ColorBar";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("ColorBar", () => {
  let wrapper;
  const handleLockStatus = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <ColorBar
        color={{ hex: "#fff", locked: false }}
        number={4}
        handleLockStatus={handleLockStatus}
      />
    );
  });
  it("should match the snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should display the lock icon if the color is locked", () => {
    const wrapper = shallow(
      <ColorBar color={{ hex: "#fff", locked: true }} number={4} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should invoke the handleLockStatus function if the '.lock-button' (while locked:false) is clicked", () => {
    const button = wrapper.find(".color-block");
    button.simulate("click");
    expect(handleLockStatus).toHaveBeenCalled();
  });
  it("should invoke the handleLockStatus function if the '.lock-button' (while locked:true) is clicked", () => {
    const wrapper = shallow(
      <ColorBar
        color={{ hex: "#fff", locked: true }}
        number={4}
        handleLockStatus={handleLockStatus}
      />
    );
    const button = wrapper.find(".color-block");
    button.simulate("click");
    expect(handleLockStatus).toHaveBeenCalled();
  });

  it("should render the hex codes vertically if the vRotate prop is passed in", () => {
    const wrapper = shallow(
      <ColorBar color={{ hex: "#fff", locked: true }} vRotate={true} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
