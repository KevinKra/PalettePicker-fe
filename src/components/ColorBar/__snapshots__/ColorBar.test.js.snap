// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBar should display the lock icon if the color is locked 1`] = `
<div
  className="color-block"
  onClick={[Function]}
  style={
    Object {
      "animation": "move 0.7s ease-in 0.5s forwards",
      "transform": "translateY(-150%)",
    }
  }
>
  <div
    className="color-bar color-5"
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottom": "5px solid black",
      }
    }
  >
    <p>
      #FFF
    </p>
  </div>
  <button
    className="lock-button"
    style={
      Object {
        "height": "15%",
        "opacity": 1,
        "visibility": "visible",
      }
    }
  >
    <i
      className="fas fa-2x fa-lock"
    />
  </button>
</div>
`;

exports[`ColorBar should match the snapshot 1`] = `
<div
  className="color-block"
  onClick={[Function]}
  style={
    Object {
      "animation": "move 0.7s ease-in 0.5s forwards",
      "transform": "translateY(-150%)",
    }
  }
>
  <div
    className="color-bar color-5"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
  >
    <p>
      #FFF
    </p>
  </div>
  <button
    className="lock-button"
    style={null}
  >
    <i
      className="fas fa-2x fa-unlock-alt"
    />
  </button>
</div>
`;

exports[`ColorBar should render the hex codes vertically if the vRotate prop is passed in 1`] = `
<div
  className="color-block"
  onClick={[Function]}
  style={
    Object {
      "animation": "move 0.7s ease-in NaNs forwards",
      "transform": "translateY(-150%)",
    }
  }
>
  <div
    className="color-bar color-NaN"
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottom": "5px solid black",
      }
    }
  >
    <p
      className="color-bar-text-vertical"
    >
      #FFF
    </p>
  </div>
  <button
    className="lock-button"
    style={
      Object {
        "height": "15%",
        "opacity": 1,
        "visibility": "visible",
      }
    }
  >
    <i
      className="fas fa-2x fa-lock"
    />
  </button>
</div>
`;
